File: DPK01-Revert-String/scala/src/DPK01_impl_1.scala
def reverseString(s: String): String = {
    var reversed = ""
    for (i <- s.length - 1 to 0 by -1) {
        reversed = reversed + s(i)
    }
    reversed
}

def main(args: Array[String]): Unit = {
    println(reverseString("Hello"))
}

File: DPK01-Revert-String/java/src/DPK01_impl_1.java
public class DPK01_impl_1 {

    public static String reverseString(String str) {
        String reversed = "";

        for (int i = str.length() - 1; i >= 0; i--) {
            reversed += str.charAt(i);
        }

        return reversed;
    }

    public static void main(String[] args) {
        String original = "Hello";
        String reversed = reverseString(original);

        System.out.println(reversed);
    }
}